Build errors (cc1: all warnings being treated as errors)

OS: Pop!_OS 22.04 LTS

Build Log
```
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Building targets ...
gcc -o src/bloomd/conn_handler.o -c -std=c99 -D_GNU_SOURCE -Wall -Wextra -Werror -O2 -pthread -Isrc/bloomd/ -Ideps/inih/ -Ideps/libev/ -Isrc/libbloom/ src/bloomd/conn_handler.c
src/bloomd/conn_handler.c: In function 'handle_client_connect':
src/bloomd/conn_handler.c:160:5: error: 'arg_buf_len' may be used uninitialized in this function [-Werror=maybe-uninitialized]
  160 |     handle_filt_key_cmd(handle, args, args_len, filtmgr_check_keys);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/bloomd/conn_handler.c:65:18: note: 'arg_buf_len' was declared here
   65 |     int buf_len, arg_buf_len, should_free;
      |                  ^~~~~~~~~~~
cc1: all warnings being treated as errors
scons: *** [src/bloomd/conn_handler.o] Error 1
scons: building terminated because of errors.
```

And once this is fixed, another error

```
scons: Reading SConscript files ...
scons: done reading SConscript files.
scons: Building targets ...
gcc -o src/bloomd/conn_handler.o -c -std=c99 -D_GNU_SOURCE -Wall -Wextra -Werror -O2 -pthread -Isrc/bloomd/ -Ideps/inih/ -Ideps/libev/ -Isrc/libbloom/ src/bloomd/conn_handler.c
gcc -o src/bloomd/bloomd.o -c -std=c99 -D_GNU_SOURCE -Wall -Wextra -Werror -O2 -pthread -Isrc/bloomd/ -Ideps/inih/ -Ideps/libev/ -Isrc/libbloom/ src/bloomd/bloomd.c
src/bloomd/bloomd.c: In function 'main':
src/bloomd/bloomd.c:180:43: error: cast between incompatible function types from 'void (*)(worker_args *)' to 'void * (*)(void *)' [-Werror=cast-function-type]
  180 |         pthread_create(&threads[i], NULL, (void*(*)(void*))worker_main, &wargs);
      |                                           ^
cc1: all warnings being treated as errors
scons: *** [src/bloomd/bloomd.o] Error 1
scons: building terminated because of errors.
```

--- a/src/bloomd/bloomd.c
+++ b/src/bloomd/bloomd.c
@@ -25,7 +25,7 @@ typedef struct {
     bloom_filtmgr *mgr;
     bloom_networking *netconf;
 } worker_args;
-static void worker_main(worker_args *args);
+static void *worker_main(worker_args *args);
 
 /**
  * By default we should run. Our signal
@@ -208,11 +208,13 @@ int main(int argc, char **argv) {
 }
 
 // Main entry point for the worker threads
-static void worker_main(worker_args *args) {
+static void *worker_main(worker_args *args) {
     // Perform the initial checkpoint with the manager
     filtmgr_client_checkpoint(args->mgr);
 
     // Enter the networking event loop forever
     start_networking_worker(args->netconf);
+
+    return NULL;
 }
 

--- a/src/bloomd/conn_handler.c
+++ b/src/bloomd/conn_handler.c
@@ -64,6 +64,7 @@ int handle_client_connect(bloom_conn_handler *handle) {
     char *buf, *arg_buf;
     int buf_len, arg_buf_len, should_free;
     int status;
+    arg_buf_len = 0;
     while (1) {
         status = extract_to_terminator(handle->conn, '\n', &buf, &buf_len, &should_free);
         if (status == -1) break; // Return if no command is available

